addon/hint/sql-hint.js

```diff
@@ -264,7 +264,7 @@
       token.string = token.string.slice(0, cur.ch - token.start);
     }
 
-    if (token.string.match(/^[.`"\w@]\w*$/)) {
+     if (token.string.match(/^[.`"\w@]+(\$?#?\w*)*/g)) {
```

Comment:
You have an extra space in front of the `if`, misaligning the indentation.

The regexp looks scary like this (nested repeat operators can easily create exponential running time). Why not just replace the `\w` with `[\w$#]`?

Reply:
updated the regex and removed space.